cmake/Helpers.cmake

```diff
@@ -0,0 +1,790 @@
+    ENDIF()
+    SET(BAKE_ICON ${BAKE_ICON} CACHE INTERNAL "")
+ENDFUNCTION()
+
+FUNCTION(CONFIGURE_VITA_SPECIFICS)
+    IF(VITA)
+        INCLUDE(platforms/vita/vita.cmake)
+    ENDIF(VITA)
+ENDFUNCTION()
+
+FUNCTION(CONFIGURE_UNIX_SPECIFICS)
+    IF(UNIX)
+        SET(CMAKE_THREAD_PREFER_PTHREAD true)
+        SET(CMAKE_THREAD_PREFER_PTHREAD ${CMAKE_THREAD_PREFER_PTHREAD} CACHE INTERNAL "")
+        FIND_PACKAGE(Threads REQUIRED)
+
+        # TODO ONLY FOR UBUNTU
+        # also put the chosen paths in the man page (Ubuntu)
+        CONFIGURE_FILE(
+            "${CMAKE_CURRENT_SOURCE_DIR}/gemrb.6.in"
+            "${CMAKE_CURRENT_BINARY_DIR}/gemrb.6"
+            IMMEDIATE @ONLY
+        )
+    ENDIF(UNIX)
+ENDFUNCTION()
+
+FUNCTION(CONFIGURE_RPI_SPECIFICS)
+    # check for RaspberryPi
+    IF(UNIX)
+        FIND_FILE(RPI NAMES bcm_host.h PATHS "/opt/vc/include")
+        # By default, Pi0 to Pi3 models use the legacy (Broadcom) GLESv2 drivers, from /opt/vc.
+        # Newer models (Pi4) don't support it, using the open source MESA drivers.
+        # NOTE: the Pi3B(+) models can also run with open source MESA drivers, but support for it must be explicitely enabled
+        IF(RPI AND NOT DISABLE_VIDEOCORE)
+            SET(HINT_GLES_LIBNAME brcmGLESv2)
+            SET(HINT_GLES_LIBNAME ${HINT_GLES_LIBNAME} CACHE INTERNAL "")
+
+            SET(HINT_GLES_INCDIR "/opt/vc/include")
+            SET(HINT_GLES_INCDIR ${HINT_GLES_INCDIR} CACHE INTERNAL "")
+
+            SET(HINT_GLES_LIBDIR "/opt/vc/lib")
+            SET(HINT_GLES_LIBDIR ${HINT_GLES_LIBDIR} CACHE INTERNAL "")
+
+            SET(OPENGL_BACKEND GLES)
+            SET(OPENGL_BACKEND ${OPENGL_BACKEND} CACHE INTERNAL "")
+
+            SET(SDL_BACKEND SDL2)
+            SET(SDL_BACKEND ${SDL_BACKEND} CACHE INTERNAL "")
+        ENDIF()
+    ENDIF()
+ENDFUNCTION()
+
+FUNCTION(CONFIGURE_TARGET_PLATFORM_SPECIFICS)
+    #EACH OF THEM DOES NOTHING IF TARGET PLATFORM IS NOT MATCHING
+    CONFIGURE_APPLE_SPECIFICS()
+    CONFIGURE_VITA_SPECIFICS()
+    CONFIGURE_UNIX_SPECIFICS()
+    CONFIGURE_RPI_SPECIFICS()
+ENDFUNCTION()
+
+FUNCTION(CONFIGURE_LINKING)
+    IF(STATIC_LINKING)
+        IF(MSVC)
+            UNSET(STATIC_LINK CACHE)
+            MESSAGE(STATUS "Static linking not (yet) supported on this platform.")
+        ELSE()
+            ADD_DEFINITIONS("-DSTATIC_LINK")
+        ENDIF()
+    ELSE()
+        IF(NOT HAVE_DLFCN_H AND NOT WIN32)
+            MESSAGE(WARNING "Dynamic linking not supported by platform, switching to static!")
+            SET(STATIC_LINK)
+            ADD_DEFINITIONS("-DSTATIC_LINK")
+            LIST(FIND CMAKE_FIND_LIBRARY_SUFFIXES ".a" FOUND)
+            IF(NOT (FOUND EQUAL "-1"))
+                LIST(APPEND CMAKE_FIND_LIBRARY_SUFFIXES ".a")
+            ENDIF()
+        ENDIF()
+    ENDIF()
+ENDFUNCTION()
+
+FUNCTION(CONFIGURE_FMT_INCLUDES)
+    # supress static analysis of fmt
+    INCLUDE_DIRECTORIES(SYSTEM ${CMAKE_CURRENT_SOURCE_DIR}/includes/fmt)
```

Comment:
Nice idea, I guess this is for clang-tidy?

Reply:
For `clangd` but yeah, it embeds `clang-tidy`.

https://clangd.llvm.org/features#clang-tidy-checks